### PR TITLE
Test: Update tcf policy version

### PR DESCRIPTION
**Pull request by marki1an**
### 🔧 Type of changes
- [ ] new bid adapter
- [ ] update bid adapter
- [ ] new feature
- [ ] new analytics adapter
- [ ] new module
- [ ] bugfix
- [ ] documentation
- [ ] configuration
- [x] tech debt (test coverage, refactorings, etc.)

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 🏎 Quality check

- [ ] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy [hunk 1]
```diff
@@ -383,7 +384,11 @@ class GdprAmpSpec extends PrivacyBaseSpec {
         then: "Bid response should contain warning"
         assert response.ext?.warnings[PREBID]*.code == [999]
         assert response.ext?.warnings[PREBID]*.message ==
-                ["Parsing consent string: ${tcfConsent} failed. TCF policy version ${invalidTcfPolicyVersion} is not supported" as String]
+                ["Unknown tcfPolicyVersion ${invalidTcfPolicyVersion}, defaulting to gvlSpecificationVersion=3" as String]
+
+        and: "Alerts.general metrics should be populated"
+        def metrics = privacyPbsService.sendCollectedMetricsRequest()
+        assert metrics["alerts.general"] == 1
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy:391]: pls also add `flushMetrics(privacyPbsService)` due to theoretical flack in test

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy [hunk 2]
```diff
@@ -26,6 +26,7 @@ import static org.prebid.server.functional.model.config.Purpose.P2
 import static org.prebid.server.functional.model.config.Purpose.P4
 import static org.prebid.server.functional.model.config.PurposeEnforcement.BASIC
 import static org.prebid.server.functional.model.config.PurposeEnforcement.NO
+import static org.prebid.server.functional.model.mock.services.vendorlist.GvlSpecificationVersion.*
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy:29]: wildcard import

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy [hunk 3]
```diff
@@ -314,7 +315,7 @@ class GdprAuctionSpec extends PrivacyBaseSpec {
         tcfPolicyVersion << [TCF_POLICY_V2, TCF_POLICY_V4, TCF_POLICY_V5]
     }
 
-    def "PBS auction should reject request with proper warning when incoming consent.tcfPolicyVersion have invalid parameter"() {
+    def "PBS auction should reject request with proper warning and metrics when incoming consent.tcfPolicyVersion have invalid parameter"() {
         given: "Tcf consent string"
         def invalidTcfPolicyVersion = PBSUtils.getRandomNumber(5, 63)
         def tcfConsent = new TcfConsent.Builder()
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy:321]: The current change should allow us to accept any value for `tcfPolicyVersion`, and we should receive `response.seatbid`. Please change `tcfConsent` to:

```
def tcfConsent = new TcfConsent.Builder()
        .setPurposesLITransparency(BASIC_ADS)
        .setTcfPolicyVersion(invalidTcfPolicyVersion)
        .setVendorLegitimateInterest([GENERIC_VENDOR_ID])
        .build()
```
Additionally, add test variants where `invalidTcfPolicyVersion` is set to `5`, random value, and max value - 63. Ensure that we include responses with request, as we no longer reject them

Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy:321]: Good catch

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy [hunk 4]
```diff
@@ -761,4 +777,35 @@ class GdprAuctionSpec extends PrivacyBaseSpec {
...
+                .build()
+
+        and: "Bid request"
+        def bidRequest = getGdprBidRequest(tcfConsent)
+
+        and: "Set vendor list response"
+        vendorListResponse.setResponse(tcfPolicyVersion)
+
+        and: "Flush metrics"
+        flushMetrics(privacyPbsService)
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy:797]: we doesn't require `flushMetrics` here

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy [hunk 5]
```diff
@@ -314,11 +314,12 @@ class GdprAuctionSpec extends PrivacyBaseSpec {
         tcfPolicyVersion << [TCF_POLICY_V2, TCF_POLICY_V4, TCF_POLICY_V5]
     }
 
-    def "PBS auction should reject request with proper warning when incoming consent.tcfPolicyVersion have invalid parameter"() {
-        given: "Tcf consent string"
-        def invalidTcfPolicyVersion = PBSUtils.getRandomNumber(5, 63)
+    def "PBS auction should reject request with proper warning and metrics when incoming consent.tcfPolicyVersion have invalid parameter"() {
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy:317]: `PBS auction shouldn't reject request with proper warning and metrics when incoming consent.tcfPolicyVersion have invalid parameter`

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy [hunk 6]
```diff
@@ -360,11 +361,12 @@ class GdprAmpSpec extends PrivacyBaseSpec {
         tcfPolicyVersion << [TCF_POLICY_V2, TCF_POLICY_V4, TCF_POLICY_V5]
     }
 
-    def "PBS amp with invalid consent.tcfPolicyVersion parameter should reject request and include proper warning"() {
-        given: "Tcf consent string"
-        def invalidTcfPolicyVersion = PBSUtils.getRandomNumber(5, 63)
+    def "PBS amp with invalid consent.tcfPolicyVersion parameter should reject request and include proper warning and metrics"() {
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAmpSpec.groovy:364]: We are no longer rejecting; we are defaulting to gvlSpecificationVersion=3

**Review by osulzhenko — CHANGES_REQUESTED** [review]

**Review by osulzhenko — APPROVED** [review]

**Pull request merged**